### ISSUE TITLE
Full Yellow Local - Fix Remoted tests in 4.2

### ISSUE THREAD

**Issue by juliamagan** [wazuh/wazuh-qa#1530]
## Description

This issue is part of the https://github.com/wazuh/wazuh-qa/issues/1516. After the research #1493, it has been detected fails and warnings in `remoted` tests for Wazuh manager. In order to sanitize current tests is necessary to fix `remoted` tests.

### Test Execution - Results

|  Server  |  **Local** | **Jenkins**  |  **Status**|
|:--:|:--: |:--:|:--:|
|**Test Execution 1**| [Results](https://github.com/wazuh/wazuh-qa/files/6746513/test_remoted_1.log)|  [Build](https://devel.ci.wazuh.info/view/Tests/job/Test_integration/879/)|    🔴  | 
|**Test Execution 2**| [Results](https://github.com/wazuh/wazuh-qa/files/6746504/test_remoted_r2.log)|  [Build](https://devel.ci.wazuh.info/view/Tests/job/Test_integration/922/) |    🔴  | 
|**Test Execution 3**| [Results](https://github.com/wazuh/wazuh-qa/files/6746503/test_remoted_r3.log) |   [Build](https://devel.ci.wazuh.info/view/Tests/job/Test_integration/879/) |    🔴  | 

### Packages details

|  Jenkins branch  | QA branch   | 
|- |- |
|  4.2 |   1516-4.2.0-full-green |

Type | Format | Architecture | Version | Revision | Tag | File name
-- | -- | -- | -- | -- | -- | --
Server | rpm | x86_64 | 4.2.0 | 40207 | v4.2.0-rc8 | wazuh-manager-4.2.0-1.1493.x86_64.rpm

### Environment 

Provider| Box| OS| CPU| Memory | 
--|--|--|--|--|
Vagrant | centos/8 | CentOS 8 | 2 | 1024 |

### Setup Local_internal_opcion.conf
```
remoted.debug=2
wazuh_database.interval=1
wazuh_db.commit_time=2
wazuh_db.commit_time_max=3
monitord.rotate_log=0
```

### Development

This fix should be developed in a branch from 4.2, and it must be merged in the epic issue full-green branch `1516-4.2.0-full-green`

| Tests |Warnings|
|--|--|
| test_remoted/test_configuration/test_basic_configuration_connection_invalid_protocol.py | 8
| test_remoted/test_configuration/test_basic_configuration_connection_valid.py | 8 
| test_remoted/test_configuration/test_basic_configuration_ipv6.py | 2
| test_remoted/test_configuration/test_basic_configuration_local_ip_valid.py | 2
| test_remoted/test_configuration/test_basic_configuration_queue_size_too_big.py | 1
| test_remoted/test_configuration/test_basic_configuration_queue_size_valid.py | 3 
| test_remoted/test_configuration/test_basic_configuration_rids_closing_time_valid.py | 8
| test_remoted/test_configuration/test_basic_configuration_syslog_allowed_denied_ips_valid.py | 5
| test_remoted/test_configuration/test_basic_configuration_syslog_denied_ips.py | 1

In order to finish this issue the following tasks should be fulfilled:

- [x] Fix `remoted` integration tests. :
  - [x] https://github.com/wazuh/wazuh-qa/issues/1570
  - [x] https://github.com/wazuh/wazuh-qa/issues/1572
  - [x] https://github.com/wazuh/wazuh-qa/issues/1573
  - [ ] https://github.com/wazuh/wazuh-qa/issues/1571 > Blocked by https://github.com/wazuh/wazuh-qa/issues/1698
  - [x] https://github.com/wazuh/wazuh-qa/issues/1578
  - [x] https://github.com/wazuh/wazuh-qa/issues/1579
  - [x] https://github.com/wazuh/wazuh-qa/issues/1577
  - [ ] https://github.com/wazuh/wazuh-qa/issues/1576  > Blocked by https://github.com/wazuh/wazuh-qa/issues/1698
  - [x] https://github.com/wazuh/wazuh-qa/issues/1575
  - [ ] https://github.com/wazuh/wazuh-qa/issues/1574
  - [x] https://github.com/wazuh/wazuh-qa/issues/1618
- [x] Full Green/ Full Yellow in remoted tests for every supported OS.
- [x] Merge `1530-full-yellow-remoted` to `1516-4.2.0-full-green`

**Comment by damarisg** [comment]
While the issues were being created, it was possible to see that the same errors message were showing in different tests.
But the method involved was different, so when starting to work, you should synchronize with those involved in the different issues that affect the same error message.

I add the list of error messages related to multiple issues.

| Error messages| Issues | 
|--|--|
| `Could not find the remoted key loading log` | https://github.com/wazuh/wazuh-qa/issues/1572 - https://github.com/wazuh/wazuh-qa/issues/1578 - https://github.com/wazuh/wazuh-qa/issues/1576 - https://github.com/wazuh/wazuh-qa/issues/1574

**Comment by damarisg** [comment]
**General Research:** Done. It allowed creating each fail contained on this issue.

**Comment by DProvinciani** [comment]
I did a full run of the remoted tests on a branch that combines the changes on the https://github.com/wazuh/wazuh-qa/pull/1625 and those on the https://github.com/wazuh/wazuh-qa/pull/1624.

The progress went much further than it was before but there were failures in a test that in fact had no failure cases in https://github.com/wazuh/wazuh-qa/issues/1573. We are going to have to take a closer look at the tests `test_remoted/test_configuration` as there seem to be some flaky tests as well. For example:

- `test_basic_configuration_connection_invalid_protocol` pass in my local environment but in issue https://github.com/wazuh/wazuh-qa/issues/1573 it is reported as failing.
- `test_basic_configuration_connection_valid` pass in my local environment but in issue https://github.com/wazuh/wazuh-qa/issues/1573 it is reported as failing.
- `test_basic_configuration_ipv6` fails in my local environment but the issue https://github.com/wazuh/wazuh-qa/issues/1573 does not report any failures.

![image](https://user-images.githubusercontent.com/5703274/126649282-ff4288cc-0655-4f7f-ba5e-e04195e17ba5.png)

**Comment by damarisg** [comment]
# UPDATE 22-07

## PRs on `1530-full-yellow-remoted`

|PRs |  
|--|
| #1624  |
| #1625  |

**Comment by DProvinciani** [comment]
### Update including changes of #1624 and #1625

I performed full `remoted` tests executions in order to determine the global status of the tests of this component including the fixes in #1624 and #1625. I got some intermittent failures that should be analyzed.

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_remoted-r1.log](https://github.com/wazuh/wazuh-qa/files/6865439/test_remoted-r1.log)| 2021-07-22 | Diego | :yellow_circle:|
|[test_remoted-r2.log](https://github.com/wazuh/wazuh-qa/files/6865442/test_remoted-r2.log)| 2021-07-22 | Diego |`test_syslog_message.py::test_syslog_message` :red_circle:|
|[test_remoted-r3.log](https://github.com/wazuh/wazuh-qa/files/6865444/test_remoted-r3.log)| 2021-07-22 | Diego |`test_agents_switching_protocols.py::test_agents_switching_protocols`:red_circle:|

**Comment by mdengra** [comment]
### 2021-07-27

Used Wazuh-QA branch: `1570-test-socket-communication`
Test results with the default settings in the `ossec.conf`: 

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_remoted_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6886149/test_remoted_local_r1.log)| 2021-07-27 | Miguel | :yellow_circle: 
|[test_remoted_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6886152/test_remoted_local_r2.log)| 2021-07-27 | Miguel | :red_circle: 
|[test_remoted_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6886153/test_remoted_local_r3.log)| 2021-07-27 | Miguel | :yellow_circle: 

The following errors happened in the second running: 
```
FAILED test_remoted/test_agent_communication/test_multi_agent_protocols_communication.py::test_multi_agents_protocols_communication[UDP_56000]
FAILED test_remoted/test_agent_communication/test_multi_agent_protocols_communication.py::test_multi_agents_protocols_communication[TCP,UDP_56000]
```
The origin of both errors is the same:
```
                        logger.error(f"Results expected: {accum_results}")
>                   raise TimeoutError(error_message)
E                   TimeoutError: Could not find the expected event in queue socket

/usr/local/lib/python3.6/site-packages/wazuh_testing/tools/monitoring.py:456: TimeoutError
```

**Comment by DProvinciani** [comment]
### 2021-07-28 Update including changes of #1662

I performed full `remoted` tests executions in order to determine the global status of the tests of this component including the fixes in #1662. I got some intermittent failures that should be analyzed.

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_remoted-r1.log](https://github.com/wazuh/wazuh-qa/files/6890733/test_remoted-r1.log)| 2021-07-28 | Diego | `test_agent_communication/test_multi_agent_protocols_communication.py` :red_circle: |
|[test_remoted-r2.log](https://github.com/wazuh/wazuh-qa/files/6890746/test_remoted-r2.log)| 2021-07-28 | Diego | :yellow_circle: |
|[test_remoted-r3.log](https://github.com/wazuh/wazuh-qa/files/6890762/test_remoted-r3.log)| 2021-07-28 | Diego | :yellow_circle: |

**Comment by damarisg** [comment]
## Update 28-07-21 

After merging #1662 the fail `test_agents_switching_protocols.py`  reported by Diego was fixed. 

I show the results:

Test Executions | Date | By | Status
-- | -- | -- | --
[ResultsAgentCommunication.log](https://github.com/wazuh/wazuh-qa/files/6892787/ResultsAgentCommunication.log) | 2021-07-22 | Seyla| 🟡
[ResultsAgentCommunication2.log](https://github.com/wazuh/wazuh-qa/files/6892788/ResultsAgentCommunication2.log) | 2021-07-22 | Seyla| 🟡
[ResultsAgentCommunication3.log](https://github.com/wazuh/wazuh-qa/files/6892789/ResultsAgentCommunication3.log) | 2021-07-22 | Seyla| 🟡

**Comment by mdengra** [comment]
### 2021-07-28

Used Wazuh-QA branch: `1530-full-yellow-remoted`
Test results with the default settings in the `ossec.conf`: 

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_remoted_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6892921/test_remoted_local_r1.log)| 2021-07-28 | Miguel | :yellow_circle: 
|[test_remoted_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6892923/test_remoted_local_r2.log)| 2021-07-28 | Miguel | :yellow_circle: 
|[test_remoted_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6892924/test_remoted_local_r3.log)| 2021-07-28 | Miguel | :yellow_circle:

**Comment by damarisg** [comment]
# Update 28-07-21
Test Results:

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[FullYellowRemoted.log](https://github.com/wazuh/wazuh-qa/files/6894962/FullYellowRemoted.log)| 2021-07-28 | Seyla| 🔴  (  https://github.com/wazuh/wazuh-qa/issues/1573)
|[FullYellowRemoted2.log](https://github.com/wazuh/wazuh-qa/files/6894963/FullYellowRemoted2.log)| 2021-07-28 | Seyla| :yellow_circle: 
|[FullYellowRemoted3.log](https://github.com/wazuh/wazuh-qa/files/6894964/FullYellowRemoted3.log)| 2021-07-28 | Seyla| 🔴   (https://github.com/wazuh/wazuh-qa/issues/1575)

**Note:**
Each test run alone works successfully and therefore is closed, but we can see that they failed in 2 random executions when I execute the Remoted cycle complete.

**Comment by mdengra** [comment]
### 2021-07-30

Used Wazuh-QA branch: `1571-fix-socket-error-in-ar-test`
Test results with the default settings in the `ossec.conf`: 

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_remoted_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6906174/test_remoted_local_r1.log)| 2021-07-30 | Miguel | :red_circle: 
|[test_remoted_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6906176/test_remoted_local_r2.log)| 2021-07-30 | Miguel | :red_circle: 
|[test_remoted_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6906177/test_remoted_local_r3.log)| 2021-07-30 | Miguel | :red_circle: 

The same test has failed in all rounds:
```
=========================== short test summary info ============================
FAILED test_remoted/test_configuration/test_basic_configuration_connection_invalid_protocol.py::test_invalid_protocol[Testing,UDP_secure_1514]
FAILED test_remoted/test_configuration/test_basic_configuration_connection_invalid_protocol.py::test_invalid_protocol[TCP,Testing_secure_1514]
FAILED test_remoted/test_configuration/test_basic_configuration_connection_invalid_protocol.py::test_invalid_protocol[TCP,UDP,Testing_secure_1514]
FAILED test_remoted/test_configuration/test_basic_configuration_connection_invalid_protocol.py::test_invalid_protocol[Testing,UDP_syslog_514]
```

Therefore I have run it standalone using the same branch:
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_basic_configuration_connection_invalid_protocol.py_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6906319/test_basic_configuration_connection_invalid_protocol.py_local_r1.log)| 2021-07-30 | Miguel | :red_circle: 
|[test_basic_configuration_connection_invalid_protocol.py_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6906321/test_basic_configuration_connection_invalid_protocol.py_local_r2.log)| 2021-07-30 | Miguel | :red_circle: 
|[test_basic_configuration_connection_invalid_protocol.py_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6906328/test_basic_configuration_connection_invalid_protocol.py_local_r3.log)| 2021-07-30 | Miguel | :red_circle: 

After confirming that errors also occur I ran it again using the `1530-full-yellow-remoted` branch:
| Test Executions | API log | Date  | By  | Status | 
|--|--|--|--|--|
|[test_basic_configuration_connection_invalid_protocol.py_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6906417/test_basic_configuration_connection_invalid_protocol.py_local_r1.log)|[api_r1.log](https://github.com/wazuh/wazuh-qa/files/6906505/api_r1.log)| 2021-07-30 | Miguel | :red_circle: 
|[test_basic_configuration_connection_invalid_protocol.py_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6906419/test_basic_configuration_connection_invalid_protocol.py_local_r2.log)|[api_r2.log](https://github.com/wazuh/wazuh-qa/files/6906509/api_r2.log)| 2021-07-30 | Miguel | :red_circle: 
|[test_basic_configuration_connection_invalid_protocol.py_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6906420/test_basic_configuration_connection_invalid_protocol.py_local_r3.log)|[api_r3.log](https://github.com/wazuh/wazuh-qa/files/6906511/api_r3.log)| 2021-07-30 | Miguel | :red_circle: 

The failure is repeated every time. Although this test appears in the issue https://github.com/wazuh/wazuh-qa/issues/1573 (case 1), the error detected is different:
```
    def get_manager_configuration(section=None, field=None):
        """Get Wazuh manager configuration response from API using GET /manager/configuration
.
.
.   
        response = requests.get(api_query, headers=api_details['auth_headers'], verify=False)
    
>       assert response.json()['error'] == 0, f"Wazuh API response status different from 0: {response.json()}"
E       AssertionError: Wazuh API response status different from 0: {'title': 'Wazuh Internal Error', 'detail': 'Timeout executing API request', 'dapi_errors': {'node01': {'error': 'Timeout executing API request', 'logfile': 'WAZUH_HOME/logs/api.log'}}, 'error': 3021}

/usr/local/lib/python3.6/site-packages/wazuh_testing/api.py:128: AssertionError
```
Since the problem seems to be API related I attach these logs.

**Comment by damarisg** [comment]
## Update 30-07-21 

It is the last execution with all changes. 

Test Executions | Date | By | Status
-- | -- | -- | --
[FullYellowRemoted.log](https://github.com/wazuh/wazuh-qa/files/6909152/FullYellowRemoted.log) | 2021-07-30 | Seyla| 🟡
[FullYellowRemoted2.log](https://github.com/wazuh/wazuh-qa/files/6909153/FullYellowRemoted2.log)| 2021-07-28 | Seyla| 🟡
[FullYellowRemoted3.log](https://github.com/wazuh/wazuh-qa/files/6909154/FullYellowRemoted3.log)| 2021-07-28 | Seyla| 🟡

**Comment by DProvinciani** [comment]
### 2021-07-30

Used Wazuh-QA branch: `1530-full-yellow-remoted`
Test results with the default settings in the `ossec.conf`: 

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_remoted-full_r1.log](https://github.com/wazuh/wazuh-qa/files/6909848/test_remoted-full_r1.log)| 2021-07-30 | Diego | :red_circle: |
|[test_remoted-full_r2.log](https://github.com/wazuh/wazuh-qa/files/6909850/test_remoted-full_r2.log)| 2021-07-30 | Diego | :yellow_circle: |
|[test_remoted-full_r3.log](https://github.com/wazuh/wazuh-qa/files/6909852/test_remoted-full_r3.log)| 2021-07-30 | Diego | :yellow_circle: |

The first round shows failures that should ve analyzed in the next tests:
```
FAILED wazuh-qa/tests/integration/test_remoted/test_agent_communication/test_multi_agent_protocols_communication.py::test_multi_agents_protocols_communication[UDP_1514]
FAILED wazuh-qa/tests/integration/test_remoted/test_socket_communication/test_syslog_message.py::test_syslog_message[syslog_UDP_514-dummy_bad_formatted_pri_header]
```

**Comment by juliamagan** [comment]
I have launched three rounds of `test_remoted/test_configuration/test_basic_configuration_connection_invalid_protocol.py` and everything seems to work properly.

### Tests results

||R1|R2|R3|Reports|
|:--:|:--:|:--:|:--:|:--:|
|CentOS 8 Manager| 🟡 | 🟡 | 🟡 |

**Comment by damarisg** [comment]
### 02-08-21

Used Wazuh-QA branch: `1530-full-yellow-remoted`


Test Executions | By | Status
-- | -- | -- |
[FullRemoted.log](https://github.com/wazuh/wazuh-qa/files/6916803/DomingoFullRemoted.log) | Seyla| 🔴 
[FullRemoted2.log](https://github.com/wazuh/wazuh-qa/files/6916804/DomingoFullRemoted2.log))| Seyla| 🔴 
[FullRemoted3.log](https://github.com/wazuh/wazuh-qa/files/6916805/DomingoFullRemoted3.log)|Seyla| 🔴

**Comment by pereyra-m** [comment]
### 2021-08-02

Wazuh-QA branch: `1530-full-yellow-remoted`

The test `test_remoted/test_socket_communication/test_syslog_message.py` was run isolated on a clean environment many times, but all the tests were successful 

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_syslog_message_1.log](https://github.com/wazuh/wazuh-qa/files/6918096/test_syslog_message_1.log)| 2021-08-02 | Matias | :yellow_circle: |
|[test_syslog_message_2.log](https://github.com/wazuh/wazuh-qa/files/6918097/test_syslog_message_2.log)| 2021-08-02 | Matias | :yellow_circle: |
|[test_syslog_message_3.log](https://github.com/wazuh/wazuh-qa/files/6918098/test_syslog_message_3.log)| 2021-08-02 | Matias | :yellow_circle: |
|[test_syslog_message_4.log](https://github.com/wazuh/wazuh-qa/files/6918099/test_syslog_message_4.log)| 2021-08-02 | Matias | :yellow_circle: |
|[test_syslog_message_5.log](https://github.com/wazuh/wazuh-qa/files/6918100/test_syslog_message_5.log)| 2021-08-02 | Matias | :yellow_circle: |
|[test_syslog_message_6.log](https://github.com/wazuh/wazuh-qa/files/6918102/test_syslog_message_6.log)| 2021-08-02 | Matias | :yellow_circle: |
|[test_syslog_message_7.log](https://github.com/wazuh/wazuh-qa/files/6918103/test_syslog_message_7.log)| 2021-08-02 | Matias | :yellow_circle: |
|[test_syslog_message_8.log](https://github.com/wazuh/wazuh-qa/files/6918104/test_syslog_message_8.log)| 2021-08-02 | Matias | :yellow_circle: |
|[test_syslog_message_9.log](https://github.com/wazuh/wazuh-qa/files/6918105/test_syslog_message_9.log)| 2021-08-02 | Matias | :yellow_circle: |
|[test_syslog_message_10.log](https://github.com/wazuh/wazuh-qa/files/6918106/test_syslog_message_10.log)| 2021-08-02 | Matias | :yellow_circle: |

**Comment by DProvinciani** [comment]
I spent some time analyzing the issues in `test_multi_agent_protocols_communication.py` reported in the next comments:

- https://github.com/wazuh/wazuh-qa/issues/1530#issuecomment-887548055
- https://github.com/wazuh/wazuh-qa/issues/1530#issuecomment-888044446
- https://github.com/wazuh/wazuh-qa/issues/1530#issuecomment-890133523
- https://github.com/wazuh/wazuh-qa/issues/1530#issuecomment-891135875

In all the executions in which we observe failures, they were caused because the Man In The Middle didn't find the event that it was looking for.

Having said that, I analyzed the MITM implementation and how it is used in these tests. Essentially for these tests the MITM will monitor the UDP socket `queue/sockets/queue`. The way in which it will monitor the queue will be:

1. Rename the `queue` to `queue.original`.
2. Read the `queue` socket from the MITM.

Considering the fact that `analysisd` also reads from that queue, the tests also stop the execution of that daemon durin the test execucion.

From this analysis we can make the next conclusions:

- The MITM implementation doesn't work as a MITM in the UDP connections. It is just a change of the receiver.
- If the `queue` receives many events, it could run into the same issue tan the FileMonitor in with it doesn't find the event before achieving the timeout.

We are analyzing the second item to see whether or not is the cause of the issue.

**Comment by pereyra-m** [comment]
### 03/08/21

The intermittent failures of `test_multi_agents_protocols_communication` may be related to a socket saturation, so the test is unable to read the required message.

Many different approaches are being carried out to read the `queue` socket, so we can probe the message is present but the test failed.
Also, even more modules are being removed from the default configuration, to reduce the messages in the socket.

**Comment by DProvinciani** [comment]
### 03/08/21

Continuing with the analysis made in https://github.com/wazuh/wazuh-qa/issues/1530#issuecomment-891932778, I modified the Wazuh testing framework to print int the console the events received by the Man In The Middle from the `queue` socket. The changes are implemented in the next branch:

https://github.com/wazuh/wazuh-qa/tree/1618-poc-mitm-monitor

After this, I performed multiples executions of the tests in order to reproduce the issue and confirm the thinking that we have related to the QueueMonitor not analyzing the events as fast as possible to avoid the timeout. We didn't manage to reproduce the issue so, the next step will be to run the tests in multiples environments in paralell.

**Comment by juliamagan** [comment]
### 04-08-21

Used Wazuh-QA branch: `1530-full-yellow-remoted`
|Test Executions  | Status
| -- | -- |
| [R1](https://github.com/wazuh/wazuh-qa/files/6930069/centos_manager_test_remoted.zip) | 🔴 |

**Comment by pereyra-m** [comment]
### 04/08/21

These are the results of the whole **remoted** tests folder but disabling all the modules including logcollector. 
This is a re-run of the tests in [this](https://github.com/wazuh/wazuh-qa/issues/1530#issuecomment-891955918) comment.

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_remoted_no_logcollector_rerun_1.log](https://github.com/wazuh/wazuh-qa/files/6931089/test_remoted_no_logcollector_rerun_1.log)| 2021-08-04| Matias |:yellow_circle:
|[test_remoted_no_logcollector_rerun_2.log](https://github.com/wazuh/wazuh-qa/files/6931088/test_remoted_no_logcollector_rerun_2.log)| 2021-08-04| Matias |:yellow_circle:
|[test_remoted_no_logcollector_rerun_3.log](https://github.com/wazuh/wazuh-qa/files/6931087/test_remoted_no_logcollector_rerun_3.log)| 2021-08-04| Matias | :yellow_circle:
|[test_remoted_no_logcollector_rerun_4.log](https://github.com/wazuh/wazuh-qa/files/6931086/test_remoted_no_logcollector_rerun_4.log)| 2021-08-04| Matias | :yellow_circle:
|[test_remoted_no_logcollector_rerun_5.log](https://github.com/wazuh/wazuh-qa/files/6931085/test_remoted_no_logcollector_rerun_5.log)| 2021-08-04| Matias | :red_circle:
|[test_remoted_no_logcollector_rerun_6.log](https://github.com/wazuh/wazuh-qa/files/6931084/test_remoted_no_logcollector_rerun_6.log)| 2021-08-04| Matias | :yellow_circle:
|[test_remoted_no_logcollector_rerun_7.log](https://github.com/wazuh/wazuh-qa/files/6931083/test_remoted_no_logcollector_rerun_7.log)| 2021-08-04| Matias | :yellow_circle:
|[test_remoted_no_logcollector_rerun_8.log](https://github.com/wazuh/wazuh-qa/files/6931082/test_remoted_no_logcollector_rerun_8.log)| 2021-08-04| Matias | :yellow_circle:
|[test_remoted_no_logcollector_rerun_9.log](https://github.com/wazuh/wazuh-qa/files/6931081/test_remoted_no_logcollector_rerun_9.log)| 2021-08-04| Matias | :yellow_circle:
|[test_remoted_no_logcollector_rerun_10.log](https://github.com/wazuh/wazuh-qa/files/6931080/test_remoted_no_logcollector_rerun_10.log)| 2021-08-04| Matias | :yellow_circle:

These results may suggest that the socket saturation is a possible cause, but still we can't be sure.

**Comment by DProvinciani** [comment]
### 04/08/21

I performed full executions of the remoted tests with some messages printed in order to have more information in case of failures. Although the issue in the `test_multi_agent_protocols_communication.py` was not reproduced, the executions showed issues in the test `test_syslog_message.py` that should be analyzed.

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[remoted-full-r1.log](https://github.com/wazuh/wazuh-qa/files/6933050/remoted-full-r1.log)| 2021-08-04| Diego|:red_circle:|
|[remoted-full-r2.log](https://github.com/wazuh/wazuh-qa/files/6933051/remoted-full-r2.log)| 2021-08-04| Diego|:red_circle:|
|[remoted-full-r3.log](https://github.com/wazuh/wazuh-qa/files/6933052/remoted-full-r3.log)| 2021-08-04| Diego|:red_circle:|
|[remoted-full-r4.log](https://github.com/wazuh/wazuh-qa/files/6933053/remoted-full-r4.log)| 2021-08-04| Diego|:red_circle:|

**Comment by damarisg** [comment]
# 04-08-21

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[FullRemoted.log](https://github.com/wazuh/wazuh-qa/files/6934141/FullRemoted.log)| 2021-08-04| Seyla|:red_circle:|
|[FullRemoted2.log](https://github.com/wazuh/wazuh-qa/files/6934142/FullRemoted2.log)| 2021-08-04| Seyla|:red_circle:|

**Comment by damarisg** [comment]
## Full Execution - JENKINS

### Results Full Remoted

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[Results1](https://ci.wazuh.info/view/Tests/job/Test_integration/9015/console)| 05-08-21| Seyla| 🟡 
|[Results2](https://ci.wazuh.info/view/Tests/job/Test_integration/9016/console)| 05-08-21| Seyla| 🟡 
|[Results3](https://ci.wazuh.info/view/Tests/job/Test_integration/9017/console)| 05-08-21| Seyla| 🟡

**Comment by pereyra-m** [comment]
### 05/08/21

Some debugging messages were added to **remoted** to verify the correct reception and sending of the test messages.
Here are the results:
- [modulesd_debug_10.log](https://github.com/wazuh/wazuh-qa/files/6939445/modulesd_debug_10.log)
- [ossec_out_10.log](https://github.com/wazuh/wazuh-qa/files/6939446/ossec_out_10.log)

It was possible to see in the failed test that **remoted** receives both messages, but fails while trying to send one of them to the socket. That is the reason why the QA framework is unable to receive the message.

More tests are being carried out to determine if the ManInTheMiddle approach of the test is responsible for the socket communication error.

#### Details

These are the messages generated by the test. The threads send the message, but only one of them gets in the queue

```
wazuh-qa/tests/integration/test_remoted/test_agent_communication/test_multi_agent_protocols_communication.py::test_multi_agents_protocols_communication[UDP_1514] 2021-08-05T13:49:28.1628171368
2021/08/05 13:49:28 wazuh-remoted[547749] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
2021/08/05 13:49:28 wazuh-remoted[547749] main.c:130 at main(): DEBUG: Wazuh home directory: /var/ossec
2021/08/05 13:49:28 wazuh-remoted[547749] main.c:148 at main(): DEBUG: This is not a worker
2021-08-05 13:49:59,001 - wazuh_testing - ERROR - Could not find the expected event in queue socket
2021-08-05 13:49:59,002 - wazuh_testing - ERROR - Results accumulated: 0
2021-08-05 13:49:59,002 - wazuh_testing - ERROR - Results expected: 1
Agent simulator sending msg: b'!11075!#AES:\x18^\xe2\xdee\xc9S\x9f\xe2F\xafp\x1b\x9dw\x8c+\xdf@\xc9>\x10iv\xd5\xa4\xffR2~\x9f\x18Y\\\xe3\xd6\xf4\x9b\xad \r\xbd-\xdcD-\xc1\x07/\xd7\x0eg\xeb\x1fJ\xfd\x80\x95\xbe\xcf\x0f\xe2h\xb5\xe6\xf7\x92R\xad\xca.\x0c\x10\xb1!\x98\xb6\r\xba\xce\x02\n\x81\xcf\xa39\x91\xac\xf8P\xc8\x96\xb9\x93l\xe4\xbe*t\xcbU\x04\\6\xb4\xa3F\x06G\x88\xd2^\xc7\x96i\x89K\x81\xf1\xd1\xc1caS\xad1\xe8\x7f'
Agent simulator sending msg: b"!11076!#AES:\xbc\xa0\xd0\xb3^\xeaZl\x10el\xa1-\x058\xf3\xd0\x9b-\xc7\x83\x9b\\\xc2\xf2GC_\xf6\xbaB1\xb6\xd0x~`\xc9\xcc\xea\xcb\xf1\x04\x85\xa9{<\xfa\xa5\xdb\xa3\xbb\xed\x05{\xde\x16\xdf\xe3D\x10\xcf\x89H\xc2\xa4\xb8\x1f\xce\n\t\x8abBUo\x9d\r'9K\x91\xcdp{&\xa3\xc7\xd3\xa6\xe5\xc6\x03\xdf@\xa3\xfaW\xc2\x0ft,\x89\xae^\xb8\xda\xee\x8d0\x89\x93\x06\xd4(3\xdc-\xddT\xd7\xabI\xbe\xaf\xc2'\x96"
Inserting item in queue: b'1:[11076] (6-2948d7f5-debian7) any->/test.log:Feb 23 17:18:20 manager sshd[40657]: test message from agent 11076\x00'
Got msg from queue: b'1:[11076] (6-2948d7f5-debian7) any->/test.log:Feb 23 17:18:20 manager sshd[40657]: test message from agent 11076\x00'
FAILED
```

Were we have the messages printed by **remoted** related to the agents 11075 and 11076

```
2021/08/05 13:49:38 wazuh-remoted[547749] secure.c:547 at HandleSecureMessage(): DEBUG: Remoted received: (1:/test.log:Feb 23 17:18:20 manager sshd[40657]: test message from agent 11075) ([11075] (5-6400c8ef-debian7) any)
2021/08/05 13:49:38 wazuh-remoted[547749] secure.c:547 at HandleSecureMessage(): DEBUG: Remoted received: (1:/test.log:Feb 23 17:18:20 manager sshd[40657]: test message from agent 11076) ([11076] (6-2948d7f5-debian7) any)
2021/08/05 13:49:38 wazuh-remoted[547749] os_net.c:506 at OS_SendUnix(): DEBUG: Sending event to queue: (1:[11076] (6-2948d7f5-debian7) any->/test.log:Feb 23 17:18:20 manager sshd[40657]: test message from agent 11076) (113) 
```
The message for the agent 11075 is never sent.

**Comment by pereyra-m** [comment]
A new issue was created with a enhancement proposal #1690.

The debugging process of the failed tests would be much faster if the required information were easily available.

**Comment by pereyra-m** [comment]
### 06/08/21

All the **remoted** tests were run after the merging the fix in #1692.
All of them passed.







| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_full_remoted_after_fix_1.log](https://github.com/wazuh/wazuh-qa/files/6948254/test_full_remoted_after_fix_1.log)| 2021-08-06| Matias | :yellow_circle:
|[test_full_remoted_after_fix_2.log](https://github.com/wazuh/wazuh-qa/files/6948255/test_full_remoted_after_fix_2.log)| 2021-08-06| Matias | :yellow_circle:
|[test_full_remoted_after_fix_3.log](https://github.com/wazuh/wazuh-qa/files/6948256/test_full_remoted_after_fix_3.log)| 2021-08-06| Matias | :yellow_circle:
|[test_full_remoted_after_fix_4.log](https://github.com/wazuh/wazuh-qa/files/6948257/test_full_remoted_after_fix_4.log)| 2021-08-06| Matias | :yellow_circle:
|[test_full_remoted_after_fix_5.log](https://github.com/wazuh/wazuh-qa/files/6948258/test_full_remoted_after_fix_5.log)| 2021-08-06| Matias | :yellow_circle:

**Comment by DProvinciani** [comment]
### 08/09/21

All the **remoted** tests were run after merging the fix in #1692.
All of them passed.

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_remoted-r1.log](https://github.com/wazuh/wazuh-qa/files/6954202/test_remoted-r1.log)| 2021-08-06 | Diego | :yellow_circle:
|[test_remoted-r2.log](https://github.com/wazuh/wazuh-qa/files/6954203/test_remoted-r2.log)| 2021-08-06 | Diego | :yellow_circle:
|[test_remoted-r3.log](https://github.com/wazuh/wazuh-qa/files/6954205/test_remoted-r3.log)| 2021-08-06 | Diego | :yellow_circle:

**Comment by damarisg** [comment]
08-09-21

All the remoted tests were run after merging the fix in #1692.

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[ResultFullRemoted.log](https://github.com/wazuh/wazuh-qa/files/6955869/ResultFullRemoted.log)| 09-08-21| Seyla| 🟡 
|[ResultFullRemoted2.log](https://github.com/wazuh/wazuh-qa/files/6955874/ResultFullRemoted2.log)|09-08-21| Seyla| 🟡
|[ResultFullRemoted3.log](https://github.com/wazuh/wazuh-qa/files/6955877/ResultFullRemoted3.log)| 09-08-21| Seyla|  🔴  `test_active_response_send_ar.py::test_active_response_ar_sending`
|[ResultFullRemoted4.log](https://github.com/wazuh/wazuh-qa/files/6955889/ResultFullRemoted4.log)| 09-08-21| Seyla |🔴 `test_multi_agent_status.py::test_protocols_communication` `test_basic_configuration_queue_size_valid.py::test_queue_size_valid`
|[ResultFullRemoted5.log](https://github.com/wazuh/wazuh-qa/files/6955901/ResultFullRemoted5.log)| 09-08-21| Seyla| 🟡 
|[ResultFullRemoted7.log](https://github.com/wazuh/wazuh-qa/files/6956068/ResultFullRemoted7.log)| 09-08-21| Seyla| 🟡 
|[ResultFullRemoted6.log](https://github.com/wazuh/wazuh-qa/files/6955909/ResultFullRemoted6.log)| 09-08-21| Seyla| 🔴 `test_multi_agent_status.py::test_protocols_communication`
|[ResultFullRemoted8.log](https://github.com/wazuh/wazuh-qa/files/6956072/ResultFullRemoted8.log)| 09-08-21| Seyla| 🔴  `test_active_response_send_ar.py::test_active_response_ar_sending`
|[ResultFullRemoted9.log](https://github.com/wazuh/wazuh-qa/files/6956073/ResultFullRemoted9.log)| 09-08-21| Seyla |🔴 `test_multi_agent_status.py::test_protocols_communication`

**Comment by damarisg** [comment]
# FINAL EXECUTION

| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_remoted-full-r1.log](https://github.com/wazuh/wazuh-qa/files/6958227/test_remoted-full-r1.log)|09-08-21| Diego| 🟡 
|[test_remoted-full-r2.log](https://github.com/wazuh/wazuh-qa/files/6958230/test_remoted-full-r2.log)|09-08-21| Diego| 🟡
|[test_remoted-full-r3.log](https://github.com/wazuh/wazuh-qa/files/6958231/test_remoted-full-r3.log)|09-08-21| Diego|  🟡
|[test_full_remoted_after_disabled_1.log](https://github.com/wazuh/wazuh-qa/files/6958501/test_full_remoted_after_disabled_1.log)| 2021-08-09| Matias |🟡
|[test_full_remoted_after_disabled_2.log](https://github.com/wazuh/wazuh-qa/files/6958502/test_full_remoted_after_disabled_2.log)| 2021-08-09| Matias | 🟡
|[test_full_remoted_after_disabled_3.log](https://github.com/wazuh/wazuh-qa/files/6958503/test_full_remoted_after_disabled_3.log)| 2021-08-09| Matias | 🟡
||10-08-21| Seyla| WIP
||10-08-21| Seyla| WIP
||10-08-21| Seyla|  WIP
||10-08-21| Seyla| WIP

**Comment by damarisg** [comment]
Test Results: 11-08-21

| Test Executions | Date  | By  | Status | Fails |
|--|--|--|--|--|
|[ResultPedroRemoted1.2.log](https://github.com/wazuh/wazuh-qa/files/6968660/ResultPedroRemoted1.2.log)| 11-08-21| Seyla| 🟡 |
|[ResultPedroRemoted1.1.log](https://github.com/wazuh/wazuh-qa/files/6968659/ResultPedroRemoted1.1.log)|11-08-21| Seyla| 🟡
|[ResultPedroRemoted.log](https://github.com/wazuh/wazuh-qa/files/6968677/ResultPedroRemoted.log)| 11-08-21| Seyla|  🟡
|[ResultPedroRemoted2.log](https://github.com/wazuh/wazuh-qa/files/6968693/ResultPedroRemoted2.log)| 11-08-21| Seyla|  🔴| `test_agent_communication/test_agent_version_shared_configuration_startup_message` 
|[ResultPedroRemoted3.log](https://github.com/wazuh/wazuh-qa/files/6968705/ResultPedroRemoted3.log)| 11-08-21| Seyla|  🔴| `test_agent_communication/test_request_agent_info.py`  (related to https://github.com/wazuh/wazuh-qa/issues/1698) `test_agent_communication/test_agent_version_shared_configuration_startup_message`

**Comment by snaow** [comment]
Closed by: https://github.com/wazuh/wazuh-qa/pull/1717
